src/SI_Toolkit/Predictors/predictor_autoregressive_GP.py

```diff
@@ -27,21 +27,26 @@
 
 
 class predictor_autoregressive_GP(predictor):
-    def __init__(self, net_name, horizon, batch_size=1, **kwargs):
-        # tf.config.run_functions_eagerly(True)
+    def __init__(self,
+                 model_name=None,
+                 path_to_model=None,
+                 horizon=None,
+                 batch_size=1,
+                 **kwargs):
+
         a = SimpleNamespace()
 
-        if '/' in net_name:
-            a.path_to_models = os.path.join(*net_name.split("/")[:-1])+'/'
-            a.net_name = net_name.split("/")[-1]
+        if path_to_model is not None:
+            a.path_to_models = path_to_model
+            a.net_name = model_name
         else:
-            a.path_to_models = PATH_TO_MODEL
-            a.net_name = net_name
+            a.path_to_models = os.path.join(*model_name.split("/")[:-1]) + '/'
+            a.net_name = model_name.split("/")[-1]
```

Comment:
`os.sep` would be a good replacement for "/" here, as you had so well used somewhere else. (learned about this today from you :) )

Reply:
Done. Although I am not sure if it works - for the moment it seems I have no working new version.